fielder: crash in HexString from negative index

**Versions**

- Go: 1.21.5
- loadgen: 0.0.4


**Steps to reproduce**

Run for a while.

**Additional context**

```
panic: runtime error: index out of range [-1]

goroutine 2088 [running]:
math/rand.(*rngSource).Uint64(...)
	math/rand/rng.go:249
math/rand.(*rngSource).Int63(0x0?)
	math/rand/rng.go:234 +0x8c
math/rand.(*Rand).Int63(...)
	math/rand/rand.go:96
math/rand.(*Rand).Int31(...)
	math/rand/rand.go:110
math/rand.(*Rand).Int31n(0x1?, 0xdb4bd40?)
	math/rand/rand.go:142 +0x8c
math/rand.(*Rand).Intn(0x0?, 0x5d63f8?)
	math/rand/rand.go:183 +0x30
main.Rng.Int(...)
	github.com/honeycombio/loadgen/fielder.go:89
main.Rng.HexString({0xf88694b949f0?}, 0x8)
	github.com/honeycombio/loadgen/fielder.go:115 +0x64
main.Rng.getValueGenerators.func14()
	github.com/honeycombio/loadgen/fielder.go:148 +0x40
main.(*Fielder).AddFields(0x40000bb0e0, {0x8886d0, 0x4014588f00}, 0x0, 0x8?)
	github.com/honeycombio/loadgen/fielder.go:479 +0x29c
main.(*SenderOTel).CreateSpan(0x20666f4c?, {0x883ee0?, 0x4006583320?}, {0x77cf26?, 0x0?}, 0x0?, 0x4?)
	github.com/honeycombio/loadgen/sender_otel_honey.go:104 +0x330
main.(*TraceGenerator).generate_spans(0x400060a000, {0x883ee0, 0x4006583320}, 0x2, 0x2, 0x1, 0x9086d489)
	github.com/honeycombio/loadgen/generator.go:98 +0x31c
main.(*TraceGenerator).generate_spans(0x400060a000, {0x883ee0, 0x4015da7740}, 0x1, 0x3, 0x18, 0x6f122eaf2)
	github.com/honeycombio/loadgen/generator.go:99 +0x24c
main.(*TraceGenerator).generate_root(0x400060a000, 0x4018acdf30?, 0x4, 0x19, 0x6fc23ac00)
	github.com/honeycombio/loadgen/generator.go:112 +0x110
main.(*TraceGenerator).generator(0x400060a000, 0x400041a000, 0x0?)
	github.com/honeycombio/loadgen/generator.go:140 +0x1ac
created by main.(*TraceGenerator).Generate in goroutine 21
	github.com/honeycombio/loadgen/generator.go:193 +0x464
```

--- a/generator.go
+++ b/generator.go
@@ -26,29 +26,29 @@ const (
 )
 
 type TraceGenerator struct {
-	depth    int
-	nspans   int
-	duration time.Duration
-	fielder  func() *Fielder
-	chans    []chan struct{}
-	mut      sync.RWMutex
-	log      Logger
-	tracer   Sender
+	depth      int
+	nspans     int
+	duration   time.Duration
+	getFielder func() *Fielder
+	chans      []chan struct{}
+	mut        sync.RWMutex
+	log        Logger
+	tracer     Sender
 }
 
 // make sure it implements Generator
 var _ Generator = (*TraceGenerator)(nil)
 
-func NewTraceGenerator(tsender Sender, fielder func() *Fielder, log Logger, opts Options) *TraceGenerator {
+func NewTraceGenerator(tsender Sender, getFielder func() *Fielder, log Logger, opts Options) *TraceGenerator {
 	chans := make([]chan struct{}, 0)
 	return &TraceGenerator{
-		depth:    opts.Format.Depth,
-		nspans:   opts.Format.NSpans,
-		duration: opts.Format.TraceTime,
-		fielder:  fielder,
-		chans:    chans,
-		log:      log,
-		tracer:   tsender,
+		depth:      opts.Format.Depth,
+		nspans:     opts.Format.NSpans,
+		duration:   opts.Format.TraceTime,
+		getFielder: getFielder,
+		chans:      chans,
+		log:        log,
+		tracer:     tsender,
 	}
 }
 
@@ -127,7 +127,7 @@ func (s *TraceGenerator) generator(wg *sync.WaitGroup, counter chan int64) {
 	s.mut.Unlock()
 
 	ticker := time.NewTicker(duration)
-	fielder := s.fielder()
+	fielder := s.getFielder()
 	defer wg.Done()
 	for {
 		select {

--- a/main.go
+++ b/main.go
@@ -159,12 +159,12 @@ func main() {
 
 	log := NewLogger(opts.DebugLevel())
 
-	fielder := func() *Fielder {
-		fielder, err := NewFielder(opts.Seed, args, opts.Format.Extra, opts.Format.Depth)
+	getFielderFn := func() *Fielder {
+		getFielder, err := NewFielder(opts.Seed, args, opts.Format.Extra, opts.Format.Depth)
 		if err != nil {
 			log.Fatal("unable to create fields as specified: %s\n", err)
 		}
-		return fielder
+		return getFielder
 	}
 
 	opts.apihost = parseHost(log, opts.Telemetry.Host, opts.Telemetry.Insecure)
@@ -220,7 +220,7 @@ func main() {
 	}()
 
 	// start the load generator to create spans and send them on the source chan
-	var generator Generator = NewTraceGenerator(sender, fielder, log, opts)
+	var generator Generator = NewTraceGenerator(sender, getFielderFn, log, opts)
 	wg.Add(1)
 	go generator.Generate(opts, wg, stop, counterChan)
 

